javascript/javascript3/week3/shoppingCart/shoppingCart.js

```diff
@@ -0,0 +1,134 @@
+const productSearch = document.getElementById('searchInput');
+let productsul = document.createElement('ul');
+document.body.appendChild(productsul);
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency) {
+      // Assuming dkr as default currency
+    const currencyApi = `https://api.currencyapi.com/v3/latest?apikey=8oj1A2di6JFwCRRghtxD1W9zWV4WcA3xVHQN1o7M&currencies=${currency}&base_currency=DKK`;
+    fetch(currencyApi)
+      .then(response => response.json())
+      .then(currencyData => {
+        const productName = document.createElement('h3');
+        const dkkPrice = document.createElement('h3');
+        const currencyPrice = document.createElement('h3');
+        const priceinCurrency = +(this.price * currencyData.data[currency].value).toFixed(3);
+        productName.innerHTML = `Product: ${this.name}`;
+        dkkPrice.innerHTML = `Price In DKK: ${this.price}`;
+        currencyPrice.innerHTML = `Price In ${currency}: ${priceinCurrency}`;
+        document.body.appendChild(productName);
+        document.body.appendChild(dkkPrice);
+        document.body.appendChild(currencyPrice);
+      });
+  }
+}
+  
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    // Implement functionality here
+    if(!this.products.includes(product)){
+      this.products.push(product);
+    }
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    if(this.products.includes(product)){
+      const productIndex = this.products.indexOf(product);
+      this.products.splice(productIndex,1);
+    } else {
+      console.log(`Product is not added/present in the products array`);
+    }
+  }
+
+  searchProduct(productName) {
+    // Implement functionality here
+      const productsCopy = this.products.map(product => product);
+      if(productName !== "") {
+          this.products = this.products.filter(product => product.name.toLowerCase().includes(productName.toLowerCase()));
+          productsul.innerHTML = "";
+          this.renderProducts();
+          this.products = productsCopy;
+      } else {
+          productsul.innerHTML = "";
+          this.renderProducts();
+      }
+  }
+
+  getTotal() {
+    // Implement functionality here
```

Comment:
"return this.products.reduce((a, b) => a + b.price, 0)"

this also can be used

Reply:
Yes, I will try it with array.reduce.